codec-http/src/main/java/io/netty/handler/codec/http/QueryStringEncoder.java

```diff
@@ -33,13 +34,17 @@
  * encoder.addParam("recipient", "world");
  * assert encoder.toString().equals("/hello?recipient=world");
  * </pre>
+ *
  * @see QueryStringDecoder
  */
 public class QueryStringEncoder {
 
-    private final String charsetName;
+    private final Charset charset;
     private final StringBuilder uriBuilder;
     private boolean hasParams;
+    private static final byte WRITE_UTF_UNKNOWN = (byte) '?';
+    private static final char[] CHAR_MAP
```

Comment:
nit: could be bit more more compact as `= "0123456789ABCDEF".toCharArray();`

Reply:
fixed